src/networkcloud/azext_networkcloud/operations/kubernetescluster/_create.py

```diff
@@ -0,0 +1,93 @@
+
+
+class Create(_Create, CustomSshOptions):
+
+    """
+    This custom code inherits from generate kubernetescluster functions. It is
+    integrated into the generated code via:
+    cli-ext/v*/ext/src/networkcloud/azext_networkcloud/commands.py
+    """
+
+    @classmethod
+    def _build_arguments_schema(cls, *args, **kwargs):
+        args_schema = super()._build_arguments_schema(*args, **kwargs)
+
+        # Build Kubernetescluster Authentication args
+        args_schema = CustomSshOptions.build_ssh_arg_schema(
+            args_schema, "AdministratorConfiguration"
+        )
+        # Build initial_agent_pool_configurations admin configuration(admin-username and ssh-key-values)
+        _element = cls._args_schema.initial_agent_pool_configurations.Element
+
+        _element.ssh_key_values = AAZListArg(
+            options=["ssh-key-values"],
+            help="The array of comma-separated SSH public keys.",
+        )
+        _element.ssh_key_values.Element = AAZStrArg()
+        # Build control_plane_node_configuration admin configuration(admin-username and ssh-key-values)
+        args_schema.control_plane_node_configuration.ssh_key_values = AAZListArg(
+            options=["ssh-key-values"],
+            help="The array of comma-separated SSH public keys.",
+        )
+        args_schema.control_plane_node_configuration.ssh_key_values.Element = (
+            AAZStrArg()
+        )
+
+        # deregister the cli arguments which users should not interact with
+        args_schema.ssh_public_keys._registered = False
+        args_schema.initial_agent_pool_configurations.Element.ssh_public_keys._registered = (
+            False
+        )
+        args_schema.control_plane_node_configuration.ssh_public_keys._registered = False
+        return args_schema
+
+    @register_callback
+    def pre_operations(self):
+        args = self.ctx.args
+        ssh_keys = []
+        ssh_keys = CustomSshOptions.add_ssh_config(args)
+        if len(ssh_keys) == 0:
+            logger.warning(
+                "No keys are selected for insertion into the Kubernetes cluster node. "
+                "The image will need to have keys or credentials "
+                "setup in order to access."
+            )
+        args.ssh_public_keys = ssh_keys
+
+        # control plane node configuration
+        ssh_keys_control_plane = []
+        ssh_keys_control_plane += CustomSshOptions.add_ssh_key_action(
+            list(args.control_plane_node_configuration.ssh_key_values)
+        )
+        args.control_plane_node_configuration.ssh_public_keys = ssh_keys_control_plane
+
+        # initial agent pool configuration
+        for x in args.initial_agent_pool_configurations:
+            ssh_keys_initial_pool = []
+            if x._data.get("ssh_key_values") is not None:
+                ssh_keys_initial_pool += CustomSshOptions.add_ssh_key_action(
+                    (x._data["ssh_key_values"].values())
+                )
```

Comment:
Please don't use `_data` property. It's internal use only.
```suggestion
            if has_value(x.ssh_key_values) and x.ssh_key_values.to_serialized_data():
                ssh_keys_initial_pool += CustomSshOptions.add_ssh_key_action(
                    (x.ssh_key_values.to_serialized_data())
                )
```

Reply:
Updated